contracts/contracts/token/OUSD.sol

```diff
@@ -208,229 +220,164 @@ contract OUSD is Initializable, InitializableERC20Detailed, Governable {
+
+        _adjustGlobals(
+            fromRebasingCreditsDiff + toRebasingCreditsDiff,
+            fromNonRebasingSupplyDiff + toNonRebasingSupplyDiff
+        );
+    }
 
-        // Credits deducted and credited might be different due to the
-        // differing creditsPerToken used by each account
-        uint256 creditsCredited = _value.mulTruncate(_creditsPerToken(_to));
-        uint256 creditsDeducted = _value.mulTruncate(_creditsPerToken(_from));
+    function _adjustAccount(address _account, int256 _balanceChange)
+        internal
+        returns (int256 rebasingCreditsDiff, int256 nonRebasingSupplyDiff)
+    {
+        RebaseOptions state = rebaseState[_account];
+        int256 currentBalance = balanceOf(_account).toInt256();
+        if (currentBalance + _balanceChange < 0) {
+            revert("Transfer amount exceeds balance");
+        }
+        uint256 newBalance = (currentBalance + _balanceChange).toUint256();
 
-        _creditBalances[_from] = _creditBalances[_from].sub(
-            creditsDeducted,
-            "Transfer amount exceeds balance"
-        );
-        _creditBalances[_to] = _creditBalances[_to].add(creditsCredited);
-
-        if (isNonRebasingTo && !isNonRebasingFrom) {
-            // Transfer to non-rebasing account from rebasing account, credits
-            // are removed from the non rebasing tally
-            nonRebasingSupply = nonRebasingSupply.add(_value);
-            // Update rebasingCredits by subtracting the deducted amount
-            _rebasingCredits = _rebasingCredits.sub(creditsDeducted);
-        } else if (!isNonRebasingTo && isNonRebasingFrom) {
-            // Transfer to rebasing account from non-rebasing account
-            // Decreasing non-rebasing credits by the amount that was sent
-            nonRebasingSupply = nonRebasingSupply.sub(_value);
-            // Update rebasingCredits by adding the credited amount
-            _rebasingCredits = _rebasingCredits.add(creditsCredited);
+        if (state == RebaseOptions.YieldDelegationSource) {
+            address target = yieldTo[_account];
+            uint256 targetOldBalance = balanceOf(target);
+            (uint256 targetNewCredits, ) = _balanceToRebasingCredits(
+                targetOldBalance + newBalance
+            );
+            rebasingCreditsDiff =
+                targetNewCredits.toInt256() -
+                creditBalances[target].toInt256();
+
+            creditBalances[_account] = newBalance;
+            creditBalances[target] = targetNewCredits;
+        } else if (state == RebaseOptions.YieldDelegationTarget) {
+            (uint256 newCredits, ) = _balanceToRebasingCredits(
+                newBalance + creditBalances[yieldFrom[_account]]
+            );
+            rebasingCreditsDiff =
+                newCredits.toInt256() -
+                creditBalances[_account].toInt256();
+            creditBalances[_account] = newCredits;
+        } else {
+            _autoMigrate(_account);
```

Comment:
Auto-migrating contracts to non-rebasing state by default penalize users of multi-signature wallets (like Gnosis Safe) by preventing them from earning yield on their OUSD holdings.

Reply:
Yes. We've have this with gnosis safes since the launch of OUSD four years ago. The difference is that the safe can opt in, while an AMM pool like Uniswap, or a third party lending market is unable to opt out.